doc/manual.rst

```diff
@@ -2108,6 +2108,11 @@ proc inside of a proc). For an inner proc an analysis is performed whether it
 accesses its environment. If it does so, it has the calling convention
 `closure`, otherwise it has the calling convention `nimcall`.
 
+In the case that a non-`iterator` procedure is explicitly specified to use the
+`closure` calling convention, but the aformentioned analysis shows that no
+environment is accessed, the calling convention is changed to `nimcall`. The
+compiler is allowed to either report a warning or error in this case.
```

Comment:
Due to generics and conditional compilation I'm not sure how great this interaction might be... I think we should start with this, but something to ponder occasionally and perhaps inspiration will strike. 😆 

I say leave this in for now, we can easily change our mind.